src/main/java/com/synopsys/integration/alert/provider/blackduck/collector/BlackDuckCollector.java

```diff
@@ -70,11 +87,11 @@ public BlackDuckCollector(final JsonExtractor jsonExtractor, final List<MessageC
     }
 
     public Optional<String> getProjectLink(final String projectVersionUrl, final String link) {
-        if (blackDuckService.isPresent() && bucketService.isPresent()) {
+        if (null != blackDuckService && null != bucketService) {
```

Comment:
We could update the constructor to set a boolean field called hasValidConnection, if either of the services are missing we could set that field to false and just check for that here.